Fix and unskip test_try_secret_request_after_settle_channel

Test is currently skipped but no reason is given:

https://github.com/raiden-network/raiden/blob/dca2128d6f8dca116bd4987cc9d71c6b875dff52/raiden/tests/fuzz/test_state_changes.py#L721-L732

Running it locally this is how the failure looks like:

```
platform linux -- Python 3.7.4, pytest-4.6.3, py-1.8.0, pluggy-0.12.0
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/home/lefteris/ew/raiden/.hypothesis/examples')
rootdir: /home/lefteris/ew/raiden, inifile: setup.cfg
plugins: select-0.1.2, forked-1.0.2, hypothesis-4.24.3, random-0.2, xdist-1.28.0
collected 1 item                                                                                                                                                                                                                                                                

raiden/tests/fuzz/test_state_changes.py F

=================================================================================================================================== FAILURES ====================================================================================================================================
_________________________________________________________________________________________________________________ test_try_secret_request_after_settle_channel __________________________________________________________________________________________________________________

    def test_try_secret_request_after_settle_channel():
        state = MultiChannelInitiatorStateMachine()
        state.replay_path = True
        state.failing_path_2 = True
    
        v1 = state.initialize(block_number=1, random=Random(), random_seed=None)
>       v2 = state.valid_init_initiator(amount=1, partner=v1, payment_id=1, secret=b"\x91" * 32)

raiden/tests/fuzz/test_state_changes.py:727: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
raiden/tests/fuzz/test_state_changes.py:355: in valid_init_initiator
    target=init_initiators,
../../.virtualenvs/raiden/lib/python3.7/site-packages/hypothesis/stateful.py:437: in rule_wrapper
    return f(*args, **kwargs)
raiden/tests/fuzz/test_state_changes.py:362: in valid_init_initiator
    assume(amount <= self._available_amount(partner))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = MultiChannelInitiatorStateMachine({}), partner_address = MultipleResults(values=(b'\x18yY]\xeaF%>\x9d\x9bi\x130\xda\x16Y\x95\xd1\x80\xd2',))

    def _available_amount(self, partner_address):
>       netting_channel = self.address_to_channel[partner_address]
E       TypeError: unhashable type: 'MultipleResults'

raiden/tests/fuzz/test_state_changes.py:341: TypeError
=========================================================================================================================== 1 failed in 0.27 seconds ===========================================================================================================================
```

--- a/raiden/tests/fuzz/test_state_changes.py
+++ b/raiden/tests/fuzz/test_state_changes.py
@@ -3,7 +3,6 @@ from copy import deepcopy
 from hashlib import sha256
 from random import Random
 
-import pytest
 from hypothesis import assume, event
 from hypothesis.stateful import (
     Bundle,
@@ -295,7 +294,6 @@ class InitiatorMixin:
         self.used_secrets = set()
         self.processed_secret_requests = set()
         self.initiated = set()
-        self.failing_path_2 = False
 
     def _action_init_initiator(self, transfer: TransferDescriptionWithSecretState):
         channel = self.address_to_channel[transfer.target]
@@ -342,8 +340,7 @@ class InitiatorMixin:
         return channel.get_distributable(netting_channel.our_state, netting_channel.partner_state)
 
     def _assume_channel_opened(self, action):
-        if not self.failing_path_2:
-            assume(self.channel_opened(action.transfer.target))
+        assume(self.channel_opened(action.transfer.target))
 
     def _is_removed(self, action):
         expiry = self.expected_expiry[action.transfer.secrethash]
@@ -714,19 +711,5 @@ def test_regression_malicious_secret_request_handled_properly():
     v2 = state.valid_init_initiator(partner=v1, amount=1, payment_id=1, secret=b"\x00" * 32)
     state.wrong_amount_secret_request(amount=0, previous_action=v2)
     state.replay_init_initator(previous_action=v2)
-
-    state.teardown()
-
-
-@pytest.mark.skip(reason="https://github.com/raiden-network/raiden/issues/5292")
-def test_try_secret_request_after_settle_channel():
-    state = MultiChannelInitiatorStateMachine()
-    state.replay_path = True
-    state.failing_path_2 = True
-
-    v1 = state.initialize(block_number=1, random=Random(), random_seed=None)
-    v2 = state.valid_init_initiator(amount=1, partner=v1, payment_id=1, secret=b"\x91" * 32)
-    state.settle_channel(partner=v1)
-    state.valid_secret_request(previous_action=v2)
 
     state.teardown()

--- a/raiden/tests/fuzz/test_state_changes.py
+++ b/raiden/tests/fuzz/test_state_changes.py
@@ -418,7 +418,7 @@ class InitiatorMixin:
             self.event("Valid SecretRequest dropped due to previous invalid one.")
         elif self._is_removed(previous_action):
             assert not result.events
-            self.event("Ohterwise valid SecretRequest dropped due to expired lock.")
+            self.event("Otherwise valid SecretRequest dropped due to expired lock.")
         else:
             assert event_types_match(result.events, SendSecretReveal)
             self.event("Valid SecretRequest accepted.")

